pkg/providers/tinkerbell/create.go

```diff
@@ -170,6 +170,19 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
 		if err := p.getHardwareFromManagementCluster(ctx, clusterSpec.ManagementCluster); err != nil {
 			return err
 		}
+
+		// for workload cluster use tinkerbell IP of the management cluster
+		managementClusterSpec, err := p.providerKubectlClient.GetEksaCluster(ctx, clusterSpec.ManagementCluster, clusterSpec.ManagementCluster.Name)
```

Comment:
Are we considering this a part of a preflight validations? Because I think we wanted to add two things

1. Default the TinkerbellIP for workload to that from management.
2. Validate to ensure that the tinkerbell IPs are match.

Reply:
We are validating tinkerbellIP constraints after overriding it inside `clusterSpecValidator`.